File: frontend/web/src/api/coordinatorDashboard.ts
/*
 * GET /api/coordinator/dashboard — агрегат для дашборда координатора
 * (admin.html view-dashboard). В одном запросе:
 *   - stats: «Активных проектов» / «Команд» / «Студентов»
 *   - attention: pending applications + unassigned students
 *   - projects: тот же shape MentorDashboardProject, но без фильтра по
 *     ментору + с заполненным polem `mentor` для отображения в meta.
 */

import { apiFetch } from './client';
import type { MentorDashboardProject } from './projects';

export interface CoordinatorDashboardStats {
  activeProjects: number;
  teams: number;
  students: number;
}

export interface CoordinatorDashboardAttention {
  pendingApplications: number;
  unassignedStudents: number;
}

export interface CoordinatorDashboardResponse {
  stats: CoordinatorDashboardStats;
  attention: CoordinatorDashboardAttention;
  projects: MentorDashboardProject[];
}

export function getCoordinatorDashboard(): Promise<CoordinatorDashboardResponse> {
  return apiFetch<CoordinatorDashboardResponse>('/coordinator/dashboard');
}

File: backend/project-service/internal/models/coordinator_dashboard.go
package models

// CoordinatorDashboardResponse — агрегат для GET /api/coordinator/dashboard.
// Pixel-port из admin.html (view-dashboard, lines 803-983):
//   - stats: 3 числовых карточки (Активных проектов / Команд / Студентов)
//   - attention: 2 алёрта (заявки + нераспределённые студенты)
//   - projects: тот же шейп что у MentorDashboardProject, но без фильтра
//     по mentor_id и с заполненным mentor-полем для отображения в meta.
type CoordinatorDashboardResponse struct {
	Stats     CoordinatorDashboardStats     `json:"stats"`
	Attention CoordinatorDashboardAttention `json:"attention"`
	Projects  []MentorDashboardProject      `json:"projects"`
}

// CoordinatorDashboardStats — числа в верхней «stats-row» полосе.
type CoordinatorDashboardStats struct {
	ActiveProjects int `json:"activeProjects"`
	Teams          int `json:"teams"`
	Students       int `json:"students"`
}

// CoordinatorDashboardAttention — счётчики для «Требует внимания».
// PendingApplications — все ожидающие утверждения заявки (новые проекты
// status='На утверждении' + change-requests с pending_proposal_data).
// UnassignedStudents — студенты с поданной заявкой, не попавшие ни в одну
// команду.
type CoordinatorDashboardAttention struct {
	PendingApplications int `json:"pendingApplications"`
	UnassignedStudents  int `json:"unassignedStudents"`
}
